include/SDL3/SDL_stdinc.h

```diff
@@ -34,34 +34,25 @@
 
 #include <SDL3/SDL_platform_defines.h>
 
-/* Most everything except Visual Studio 2013 and earlier has stdbool.h now */
-#if defined(_MSC_VER) && (_MSC_VER < 1910)
-#define SDL_DEFINE_STDBOOL
-#endif
-/* gcc-2.95 had non-standard stdbool.h */
-#if defined(__GNUC__) && (__GNUC__ < 3)
-#define SDL_DEFINE_STDBOOL
-#endif
+#include <stdarg.h>
+#include <stdint.h>
+#include <string.h>
+#include <wchar.h>
 
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 #include <inttypes.h>
 #endif
-#include <stdarg.h>
+
 #ifndef __cplusplus
-#ifdef SDL_DEFINE_STDBOOL
-#ifndef __bool_true_false_are_defined
-#define __bool_true_false_are_defined 1
-#define bool  uint8_t
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
```

Comment:
This is not necessarily a good condition. Compiler can be in C89 mode but has extensions not turned off. IMO, the old version was better where we handled known-broken compilers.

Reply:
I'm kind of inclined to leave it, since if someone wants to use extensions, they can just include stdbool.h themselves and SDL will use that definition.

@Sackzement, does this work in your environment?

Reply:
One notable problem with this is that it can prevent using `_Bool` and `bool` interchaningly. E.g.:
```c
#include <SDL3/SDL.h>
bool foo;
_Bool* bar (void) {
  return &foo;
}
```

```
gcc -Wall -I. -c 1.c
1.c: In function ‘bar’:
1.c:5: warning: return from incompatible pointer type
```

Reply:
What a pain. Either we include stdbool.h and we get warnings that _Bool is a non-standard extension, or we don't include stdbool.h, and we get warnings if someone tries to use _Bool.

One more try...